shared/datatypes/state_machine.go

```diff
@@ -280,6 +280,13 @@ func (sm *StateMachine) GetMemory(in *Msg, k string) Memory {
 	return Memory{Key: k, Val: buf, logger: sm.logger}
 }
 
+// DeleteMemory deletes a memory for a given key. It is not an error to delete a
+// key that does not exist.
+func (sm *StateMachine) DeleteMemory(in *Msg, k string) {
+	q := `DELETE FROM states WHERE userid=$1 AND pluginname=$2 AND key=$3`
+	sm.db.Exec(q, in.User.ID, sm.pluginName, k)
```

Comment:
To be consistent with the other Memory functions, we should be checking the value here and logging errors. See https://github.com/eviltoylet/abot/blob/master/shared/datatypes/state_machine.go#L260